server/src/types/DocumentRepresentation/BoundingBox.ts

```diff
@@ -142,21 +153,20 @@ export class BoundingBox {
     const intersectionRectangleTop: number = Math.max(box1.top, box2.top);
 
     if (
-      intersectionRectangleRight < intersectionRectangleLeft &&
-      intersectionRectangleBottom < intersectionRectangleTop
+      intersectionRectangleRight > intersectionRectangleLeft &&
+      intersectionRectangleBottom > intersectionRectangleTop
```

Comment:
I think this should be:
`intersectionRectangleRight > intersectionRectangleLeft || intersectionRectangleBottom > intersectionRectangleTop`

( || instead of &&)

Reply:
Height **and** width should both be positive in order to get a positive area.

Reply:
Refactored for clarity:
https://github.com/axa-group/Parsr/blob/b16661c59b54417a1861d69836f8a7bcb47b6dc6/server/src/types/DocumentRepresentation/BoundingBox.ts#L156-L158

Reply:
You're right! I was thinking about the conditions for the boxes to not intersect